### ISSUE TITLE
View Transitions API does not work properly in Astro version 3.0.13.

### ISSUE THREAD

**Issue by jsparkdev** [withastro/astro#8557]
### Astro Info

```block
Astro                    v3.0.13
Node                     v20.6.1
System                   macOS (arm64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             @astrojs/react
                         @astrojs/tailwind
```

### If this issue only occurs in one browser, which browser is a problem?

Chrome

### Describe the Bug

The View Transitions API does not work properly in production mode in Astro version 3.0.13. It works fine in development mode.

I encountered this problem, so I downgraded the version to 3.0.12 and it now works fine in production mode.

If I remove the code added to `ViewTransitions.astro` in PR #8472  it works fine, but I don't know what other problems this may cause.

You can check it at the stackblitz address below.

### What's the expected result?

I hope that the View Transition API will work properly in production mode in Astro 3.0.13 version.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-jpf4hk?file=src%2Fpages%2Findex.astro,src%2Flayout%2FLayout.astro,src%2Fpages%2Fother.astro,package.json&on=stackblitz

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by zkMake** [comment]
I can confirm this bug. Had the same issue after upgrading to 3.0.13.

With 3.0.13 view transitions don't work.
Downgrading back to 3.0.12, view transitions start working again.

This happens only in the production build. Dev mode seems to be working fine.

**Comment by matthewp** [comment]
This is fixed in 3.1, thanks!

**Comment by zkMake** [comment]
@matthewp, I can confirm the issue is resolved in 3.1.0

Also, I believe the original report of the issue by @jsparkdev clearly states that the ViewTransitions api is not working as expected in production mode in 3.0.13 while also providing a minimal reproduction on Stackblitz.

**Comment by jsparkdev** [comment]
After updating Astro to 3.1.0, I can confirm that this bug has been fixed. thank you @zkMake  @matthewp !
